onnxscript/rewriter/llama_rule_sets.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+import onnxscript.ir as ir
+import onnxscript.rewriter.no_op as no_op
+import onnxscript.rewriter.pattern as orp
+from onnxscript.rewriter import pattern
+
+_op = pattern.onnxop
+
+
+def transpose_identity(x, perm):
+    return _op.Transpose(x, perm=perm)
+
+
+def transpose_identity_check(x: ir.Value, perm: ir.Attr | ir.RefAttr) -> bool:
+    if isinstance(perm, ir.RefAttr):
+        return False
+    if perm.type == ir.AttributeType.INTS:
+        if perm.value == list(range(len(perm.value))):
+            return True
+    return False
+
+
+def transpose_identity_rewrite(op, x: ir.Value, perm: ir.Attr | None = None):
+    return op.Identity(x)
+
+
+def transpose_transpose(x, perm1, perm2):
+    return _op.Transpose(_op.Transpose(x, perm=perm1), perm=perm2)
+
+
+def transpose_transpose_check(
+    x: ir.Value, perm1: ir.Attr | ir.RefAttr, perm2: ir.Attr | ir.RefAttr
+) -> bool:
+    if isinstance(perm1, ir.RefAttr) or isinstance(perm2, ir.RefAttr):
+        return False
+    return True
+
+
+def _apply_transpose(perm: tuple[int, ...], on: list[int]) -> list[int]:
```

Comment:
Just wondering what `on` is and where it comes from?

Reply:
apply transpose ... on ...